BibBuddy/app/src/main/java/de/bibbuddy/ExportBibTex.java

```diff
@@ -0,0 +1,175 @@
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The ExportBibTex creates and writes a BibTex file.
+ * It contains methods for generation of contents used
+ * for the export of the BibTex file.
+ *
+ * @author Silvia Ivanova
+ */
+
+public class ExportBibTex {
+
+  private final String folderName;
+  private final String fileName;
+
+  private final String bibFileType = ".bib";
+
+  /**
+   * The ExportBibTex is responsible for the creating, writing
+   * and retrieving of contents, needed for the BibTex Export.
+   *
+   * @param folderName name of the folder in which the
+   *                   BibTex file should be stored
+   * @param fileName name of the BibTex file
+   */
+  public ExportBibTex(String folderName, String fileName) {
+    this.folderName = folderName;
+    this.fileName = fileName;
+  }
+
+  /**
+   * Creates a BibTex file.
+   */
+  public void createBibFile() {
+    String rootPathStr = Environment.getExternalStorageDirectory() + File.separator
+        + folderName + File.separator + fileName + bibFileType;
+
+    try {
+      File file = new File(rootPathStr);
+
+      if (file.exists()) {
+        file.delete();
+      }
+
+      file.createNewFile();
+
+      FileOutputStream out = new FileOutputStream(file);
+      out.flush();
+      out.close();
+
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Generates the BibTex format for all books in a given shelf.
+   *
+   * @param shelfId id of the shelf
+   * @param bookDao object of the BookDao class
+   *                Depends on the used fragment.
+   *                Can be accessed through BookFragment or LibraryModel.
+   * @param noteDao object of the class NoteDao
+   *                Depends on the used fragment.
+   *                Can be accessed through BookFragment, LibraryModel or
+   *                NoteModel.
+   */
+  public String getBibFormatBook(Long shelfId, BookDao bookDao, NoteDao noteDao) {
+
+    List<Long> bookIdsCurrentShelf =
+        bookDao.getAllBookIdsForShelf(shelfId);
+    String bibFormat = "";
+
+    // for each book in the current shelf
+    for (int i = 0; i < bookIdsCurrentShelf.size(); i++) {
+      Book book;
+      book = bookDao.findById(bookIdsCurrentShelf.get(i));
+      Long bookId = bookIdsCurrentShelf.get(i);
+
+      bibFormat =  "@book{" + getBibKey(book)
+          + "isbn={" + book.getIsbn() + "}," + '\n'
+          + getBibAuthorNames(bookId, bookDao)
+          + "title={" + book.getTitle() + "}," + '\n'
+          + "subtitle={" + book.getSubtitle() + "}," + '\n'
+          + "volume={" + book.getVolume() + "}," + '\n'
+          + "publisher={" + book.getPublisher() + "}," + '\n'
+          + "edition={" + book.getEdition() + "}," + '\n'
+          + getBibNotes(book, noteDao)
+          + "year=" + book.getPubYear() + '\n' + "}" + '\n' + '\n';
+
+    }
+    return bibFormat;
```

Comment:
Please add a newline before a return for better readability
```suggestion

    return bibFormat;
```